### PR TITLE
resolved: pass mDNS reply packets to each transaction exactly once

### PR DESCRIPTION
**Pull request by poettering**
Previously we'd iterate through the RRs of an mDNS reply and then find
exactly one matching transaction on our scope for it, and pass it as
reply to that. If multiple RRs of the same packet match we'd pas the
packet multiple times to the transaction even.

This all doesn't really work anymore since there can be multiple open
transactions for the same key (with different flags), and it's kinda
ugly anywy. Hence let's turn this around: let's iterate through the
transactions and check if any of the included RRs match it, and if so
pass the packet to that transaction exactly once.

This speeds up mDNS a bit, since previously we'd oftentimes fail to find
all suitable transactions for an mDNS reply (because there can be
multiple transactions for the same RR key with different flags, and we
checked exactly one flag combination). Which would then mean the
transaction would time out, and be retried – at which point the cache
would be populated and thus it would still succeed, but only after this
timeout. With this fix this is corrected: every transaction that matches
will get the reply, instantly as we get it.

### CODE REVIEW

File: src/resolve/resolved-mdns.c [hunk 1]
```diff
@@ -265,6 +265,7 @@ static int on_mdns_packet(sd_event_source *s, int fd, uint32_t revents, void *us
```
Comment (yuwata) [inline src/resolve/resolved-mdns.c:265]: Not a matter, as we usually ignore typo in commit message, `we'd pas` -> `we'd pass`

**Review by yuwata — APPROVED** [review]
LGTM, though I have not tested yet. Are there any linked issue or reproducers for this?

**Review by bluca — APPROVED** [review]
LGTM too

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**